### ISSUE TITLE
[iOS] FlexLayout is not being rendered correctly in columns

### ISSUE THREAD

**Issue by jimbobbennett** [xamarin/Xamarin.Forms#2741]
### Description
When using a FlexLayout on a page with two items in a column - using the Holy Grail style layout, the top part of the layout is not visible on iOS

### Steps to Reproduce

1. Open the code from [MobileAppsOfTomorrow-master.zip](https://github.com/xamarin/Xamarin.Forms/files/2009088/MobileAppsOfTomorrow-master.zip)
The code in question is in `MainPage.xaml`:
```xml
    <FlexLayout Direction="Column">
        <Grid BackgroundColor="{StaticResource MicrosoftBlue}" 
              Padding="10">
            <Grid.ColumnDefinitions>
                <ColumnDefinition Width="*"/>
                <ColumnDefinition Width="Auto"/>
                <ColumnDefinition Width="*"/>
                <ColumnDefinition Width="Auto"/>
                <ColumnDefinition Width="*"/>
            </Grid.ColumnDefinitions>
            <Image Grid.Column="1"
                   Source="TakePhoto" 
                   Aspect="AspectFit"/>
            <Image Grid.Column="3"
                   Source="SelectFromLibrary" 
                   Aspect="AspectFit" 
                   Margin="10"/>
        </Grid>
        <ListView FlexLayout.Grow="1"
                  SeparatorColor="Transparent"
                  BackgroundColor="Green">      
            <ListView.ItemTemplate>
                <DataTemplate>
                    <ViewCell>
                        <Frame />
                    </ViewCell>
                </DataTemplate>
            </ListView.ItemTemplate>
        </ListView>
    </FlexLayout>
```
2. Run the app on Andoid
3. Run the app on iOS

### Expected Behavior

On both platforms there should be a blue grid at the top with two images showing.

### Actual Behavior

On Android this works correctly. On iOS there blue grid is not shown except on the iPhone SE simulator which shows it as a very small blue bar

### Basic Information

- Version with issue:
- Last known good version: No idea - first time I've tried this code
- IDE: VS for Mac
- Platform Target Frameworks: 
  - iOS:   11.11.0.280
  - Android: 8.3.0.19
- Nuget Packages:
  - Xamarin Forms 3.0.0.482510 and 3.1.0.469394-pre1
- Affected Devices:
  - All iOS devices

### Screenshots

Android:

![screenshot-1526474380800](https://user-images.githubusercontent.com/1710385/40117358-aa6495fa-590e-11e8-9f52-64f99b3a4ba6.jpg)

iOS iPhoneX:

![simulator screen shot - iphone x - 2018-05-16 at 13 37 06](https://user-images.githubusercontent.com/1710385/40117374-b3de8d52-590e-11e8-933e-5c6b11b86a5d.png)

iOS iPhone SE:

![simulator screen shot - iphone se - 2018-05-16 at 13 41 19](https://user-images.githubusercontent.com/1710385/40117424-d8d81812-590e-11e8-825d-56fbec7a3eab.png)

### Reproduction Link

[MobileAppsOfTomorrow-master.zip](https://github.com/xamarin/Xamarin.Forms/files/2009088/MobileAppsOfTomorrow-master.zip)

**Comment by pauldipietro** [comment]
The reproduction does exhibit the behavior on a 5s simulator.

**Comment by StephaneDelcroix** [comment]
tl;dr 

set `FlexLayout.Shrink="0"` to your `Grid`

long answer:

Both the grid, and the LV, have a `shrink` value of `1` (the default). the `FlexLayout` asks both what they'd want as vertical space. The `Grid` answers with `66` (with an image size of `26`, which is the image, the margin, the padding), and the `ListView` request `812` (on an iPhoneX). The `ListView` always tries to occupy all the space, and always request the maximum possible, knowing that it'll be shrunk, but that doesn't matter because it's a scrollable content (see e.g. #2531).

The `FlexLayout` then computes available space for each control, and reduce both by `33`, as they both have the same `shrink` value.

The fact that the top bar doesn't appear at all on iphoneX looks like a bug that has been solved since (I ran my experiment on the `3.1.0` branch, and I indeed get a 33px bar (hidden behind the notch).
